packages/flutter/lib/src/widgets/image.dart

```diff
@@ -65,17 +65,38 @@ ImageConfiguration createLocalImageConfiguration(BuildContext context, { Size si
 /// The [BuildContext] and [Size] are used to select an image configuration
 /// (see [createLocalImageConfiguration]).
 ///
+/// The `onError` argument can be used to manually handle errors while precaching.
+///
 /// See also:
 ///
 ///   * [ImageCache], which holds images that may be reused.
-Future<Null> precacheImage(ImageProvider provider, BuildContext context, { Size size }) {
+Future<Null> precacheImage(
+  ImageProvider provider,
+  BuildContext context, {
+  Size size,
+  ImageErrorListener onError,
+}) {
   final ImageConfiguration config = createLocalImageConfiguration(context, size: size);
   final Completer<Null> completer = new Completer<Null>();
   final ImageStream stream = provider.resolve(config);
   void listener(ImageInfo image, bool sync) {
     completer.complete();
   }
-  stream.addListener(listener);
+  void errorListener(dynamic exception, StackTrace stackTrace) {
```

Comment:
I'm not familiar enough with the Flutter style to know if this is typical so asking to learn:
This named function definition inside a function feels weird to me, I'm not completely sure how to make it look less weird, maybe make it an anonymous closure defined inside the parameters list of addListener?
(Again this is more a question for me to learn then an AI for you)

Reply:
Or maybe:
`ImageErrorListener errorListener = (...) {...}`

?

[again, just asking here]

Reply:
we don't use nested functions much, but they're fine. I agree they look weird. They are capturing, which is worth worrying about.

Reply:
I don't really have a strong preference. All 3 options feel more or less the same to me in readability. I just kept the existing pattern. I'm happy to change to any other variants if you guys have a strong preference.

Capturing is intentional here. Wouldn't an anonymous or a closure variable capture as well?

Reply:
It's fine that they're capturing, I just mean that it's something to bear in mind (not all languages with nested functions capture, so it's not necessarily obvious).